src/test/java/fi/metatavu/metamind/test/functional/builder/impl/IntentTestBuilderResource.java

```diff
@@ -0,0 +1,204 @@
+package fi.metatavu.metamind.test.functional.builder.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+
+import javax.ws.rs.DELETE;
+
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryCreated;
+import org.json.JSONException;
+
+import com.sun.tools.corba.se.idl.constExpr.Not;
+
+import feign.FeignException;
+import fi.metatavu.metamind.ApiClient;
+import fi.metatavu.metamind.client.IntentsApi;
+import fi.metatavu.metamind.client.model.IntentType;
+import fi.metatavu.metamind.client.model.Knot;
+import fi.metatavu.metamind.client.model.Intent;
+import fi.metatavu.metamind.client.model.IntentTrainingMaterials;
+import fi.metatavu.metamind.client.model.Story;
+import fi.metatavu.metamind.client.model.TrainingMaterial;
+import fi.metatavu.metamind.client.model.TrainingMaterialType;
+import fi.metatavu.metamind.rest.api.TrainingMaterialsApi;
+import fi.metatavu.metamind.test.functional.builder.AbstractTestBuilderResource;
+import fi.metatavu.metamind.test.functional.builder.TestBuilder;
+import io.reactivex.internal.operators.completable.CompletableCreate;
+
+public class IntentTestBuilderResource extends AbstractTestBuilderResource<Intent, IntentsApi> {
+
+  private Map<UUID, UUID> intentStoryMap = new HashMap<>();
+  
+  public IntentTestBuilderResource(TestBuilder testBuilder, ApiClient apiClient) {
+    super(testBuilder, apiClient);
+  }
+  
+  /**
+   * Creates a new intent
+   * 
+   * @param storyId storyId UUID
+   * @param sourceKnot knot 
+   * @param targetKnot knot 
+   * @param name intent name
+   * @return
+   */
+
+  public Intent create(UUID storyId, Knot sourceKnot, Knot targetKnot, String name) {
+    IntentTrainingMaterials intentTrainingMaterials = new IntentTrainingMaterials();
+    
+    Intent intent = new Intent();
+    intent.setSourceKnotId(sourceKnot.getId());
+    intent.setTargetKnotId(targetKnot.getId());
+    intent.setName(name);
+    intent.setType(IntentType.NORMAL);
+    intent.setGlobal(false);
+    intent.setQuickResponse("Quick response");
```

Comment:
type, global, quickResponse and quickResponseOrder should be passed as parameters so they could be tested also